tests/integration/test_asynchronous_metrics_pk_bytes_fields/test.py

```diff
@@ -0,0 +1,131 @@
+        res1 = a() if callable(a) else a
+        res2 = b() if callable(b) else b
+        if condition(res1, res2):
+            return res1, res2
+        retries_done += 1
+        if retries_done >= retries or (time.time() - start_time) > timeout:
+            return res1, res2
+        time.sleep(delay)
+
+
+def test_total_pk_bytes_in_memory_fields(started_cluster):
+    query_create = """CREATE TABLE test_pk_bytes
+    (
+       a UInt64,
+       b UInt64
+    )
+    Engine=MergeTree()
+    ORDER BY a SETTINGS index_granularity=1"""
+    node.query(query_create)
+
+    query_pk_bytes = "SELECT value FROM system.asynchronous_metrics WHERE metric = 'TotalPrimaryKeyBytesInMemory';"
+    query_pk_bytes_allocated = """SELECT value FROM system.asynchronous_metrics 
+                                  WHERE metric = 'TotalPrimaryKeyBytesInMemoryAllocated';"""
+
+    # query for metrics before inserting anything into the table
+    pk_bytes_before = int(node.query(query_pk_bytes).strip())
+    pk_bytes_allocated_before = int(node.query(query_pk_bytes_allocated).strip())
+
+    # insert data into the table and select
+    node.query(
+        """INSERT INTO test_pk_bytes SELECT number + 20, number * 20 from numbers(1000000)"""
+    )
+
+    node.query("""SELECT * FROM test_pk_bytes where a > 1000000""")
+
+    # functions to query primary key bytes used and allocated in memory
+    def res_pk_bytes():
+        return int(node.query(query_pk_bytes).strip())
+
+    def res_pk_bytes_allocated():
+        return int(node.query(query_pk_bytes_allocated).strip())
+
+    # query again after data insertion (make a reasonable amount of retries)
+    # metrics should be greater after inserting data
+    pk_bytes_before, pk_bytes_after = query_until_condition(
+        pk_bytes_before, res_pk_bytes, condition=greater
+    )
+    assert pk_bytes_after > pk_bytes_before
+
+    pk_bytes_allocated_before, pk_bytes_allocated_after = query_until_condition(
+        pk_bytes_allocated_before, res_pk_bytes_allocated, condition=greater
+    )
+    assert pk_bytes_allocated_after > pk_bytes_allocated_before
+
+    # insert some more data
+    node.query(
+        """INSERT INTO test_pk_bytes SELECT number + 100, number * 200 from numbers(1000000)"""
+    )
+    node.query("""SELECT * FROM test_pk_bytes""")
+
+    # query again and compare the metrics.
+    # metrics should be greater after inserting more data
+    pk_bytes_after, pk_bytes_after_2 = query_until_condition(
+        pk_bytes_after, res_pk_bytes, condition=greater
+    )
+    assert pk_bytes_after_2 > pk_bytes_after
+
+    pk_bytes_allocated_after, pk_bytes_allocated_after_2 = query_until_condition(
+        pk_bytes_allocated_after, res_pk_bytes_allocated, condition=greater
+    )
+    assert pk_bytes_allocated_after_2 > pk_bytes_allocated_after
+
+    # alter the table to drop some data
+    node.query(
+        "ALTER table test_pk_bytes DELETE where a < 1000000 SETTINGS mutations_sync=1;"
```

Comment:
Lest change this query. It is a bad example how to drop all the data in clickhouse.
`truncate table test_pk_bytes` is most efficient.

Reply:
I'm was dropping some data but not all data here which is why I used ALTER. But, we can surely truncate the table too and I have updated the test.